src/model/SplineSegment.cpp

```diff
@@ -0,0 +1,76 @@
+#include "SplineSegment.h"
+
+#include <cmath>
+
+SplineSegment::SplineSegment(const Point& p, const Point& q):
+        firstKnot(p), firstControlPoint(p), secondKnot(q), secondControlPoint(q) {}
+
+SplineSegment::SplineSegment(const Point& p, const Point& fp, const Point& sp, const Point& q) {
+    firstKnot = p;
+    firstControlPoint = fp;
+    secondControlPoint = sp;
+    secondKnot = q;
+}
+
+void SplineSegment::draw(cairo_t* cr) {
+    cairo_move_to(cr, firstKnot.x, firstKnot.y);
+    cairo_curve_to(cr, firstControlPoint.x, firstControlPoint.y, secondControlPoint.x, secondControlPoint.y,
+                   secondKnot.x, secondKnot.y);
+}
+
+auto SplineSegment::toVectorOfPoints() -> std::vector<Point> {
+    std::vector<Point> pointArray = {};
+    if (isFlatEnough()) {
+        pointArray = {firstKnot, secondKnot};
+    } else {
+        std::pair<SplineSegment, SplineSegment>&& twoSegments = subdivide(0.5);
+        pointArray = std::move(twoSegments.first.toVectorOfPoints());
+        pointArray.pop_back();
+        std::vector<Point>&& secondArray = twoSegments.second.toVectorOfPoints();
+        pointArray.insert(pointArray.end(), std::make_move_iterator(secondArray.begin()),
+                          std::make_move_iterator(secondArray.end()));
+    }
+    return pointArray;
+}
+
+auto SplineSegment::subdivide(float t) -> std::pair<SplineSegment, SplineSegment> {
```

Comment:
This function can be const:
```suggestion
auto SplineSegment::subdivide(float t) const -> std::pair<SplineSegment, SplineSegment> {
```